src/olympia/shelves/forms.py

```diff
@@ -30,24 +31,27 @@ def clean(self):
         base_url = settings.INTERNAL_SITE_URL
 
         endpoint = data.get('endpoint')
+        addon_type = data.get('addon_type')
         criteria = data.get('criteria')
 
-        if criteria is None:
+        if None in (addon_type, criteria):
```

Comment:
it's correct but I think `if addon_type is None or criteria is None:` is more explicit.

Reply:
But is that what we actually want?  I.e. can `addon_type` ever be `None` and the form be valid?

Reply:
Ahh, the default is set to 'extension' and the field will always have a selected option, so `addon_type` can never be set to `None`.